nptdms/channel_data.py

```diff
@@ -59,7 +56,8 @@ def append_data(self, data):
 
     @property
     def data(self):
-        return np.array(self._data, dtype=self._dtype)
+        dtype = self._dtype if len(self._data) > 0 else np.dtype('void')
```

Comment:
This doesn't look right to me, it should be valid to have channels with length 0 that have a non-void dtype. Maybe this should be?
```suggestion
        dtype = self._dtype if self._dtype is not None else np.dtype('void')
```

Reply:
Correct, good catch. Fixed in e53f64a3f49d6df56a80d56445a14ed7a285c580.

I also realized I had to switch `np.dtype('void')` in most places to `np.dtype('V8')`, as when you actually instantiate an array with `dtype('void')`, the actual type of the array is the concrete `V8` type.